bridges/LeBonCoinBridge.php

```diff
@@ -14,176 +13,211 @@ class LeBonCoinBridge extends BridgeAbstract {
-						'Équipement bébé' => 'equipement_bebe',
-						'Vêtements bébé' => 'vetements_bebe'
+						 'Tous' => '18',
+						 'Ameublement' => '19',
+						 'Electroménager' => '20',
+						 'Arts de la table' => '45',
+						 'Décoration' => '39',
+						 'Linge de maison' => '46',
+						 'Bricolage' => '21',
+						 'Jardinage' => '52',
+						 'Vêtements' => '22',
+						 'Chaussures' => '53',
+						 'Accessoires & Bagagerie' => '47',
+						 'Montres & Bijoux' => '42',
+						 'Equipement bébé' => '23',
+						 'Vêtements bébé' => '54',
 					),
-					'AUTRES' => 'autres'
+					'AUTRES' => '37'
+				)
+			),
+			'o' => array(
+				'name' => 'Vendeur',
+				'type' => 'list',
+				'values' => array(
+					'Tous' => '',
+					'Particuliers' => 'private',
+					'Professionnels' => 'pro',
 				)
 			)
 		)
 	);
 
 	public function collectData(){
 
-		$category = $this->getInput('c');
-		if(empty($category)) {
-				$category = 'annonces';
+		$params = array(
+			'text' => $this->getInput('k'),
+			'region' => $this->getInput('r'),
+			'category' => $this->getInput('c'),
+			'owner_type' => $this->getInput('o'),
+		);
+
+		$url = self::URI . 'rechereche/?' . http_build_query($params);
+		$headers = get_headers($url);
+
+		if(substr($headers[0], 9, 3) != "200") {
+			returnServerError('Could not request LeBonCoin. Tried: ' . $url);
+		} else {
+			$html = file_get_contents($url);
+		}
+
+		if(!preg_match('/^<script>window.FLUX_STATE[^\r\n]*/m', $html, $matches)) {
+			returnServerError('Could not parse JSON in page content.');
 		}
 
-		$html = getSimpleHTMLDOM(self::URI
-		. $category
-		. '/offres/'
-		. $this->getInput('r')
-		. '/?f=a&th=1&q='
-		. urlencode($this->getInput('k')))
-			or returnServerError('Could not request LeBonCoin.');
+		$clean_match = str_replace(
+			array('</script>', '<script>window.FLUX_STATE = '),
+			array('', ''),
+			$matches[0]
+		);
+		$json = json_decode($clean_match);
 
-		$list = $html->find('.tabsContent', 0);
-		if($list === null) {
+		if($json->adSearch->data->total === 0) {
 			return;
 		}
 
-		$tags = $list->find('li');
+		foreach($json->adSearch->data->ads as $element) {
+
+			$item['title'] = $element->subject;
+			$item['content'] = $element->body;
+			$item['date'] = $element->index_date;
+			$item['timestamp'] = strtotime($element->index_date);
+			$item['uri'] = $element->url;
+			$item['ad_type'] = $element->ad_type;
+			$item['author'] = $element->owner->name;
+
+			if(isset($element->city)) {
+				$item['city'] = $element->city;
+			}
 
-		foreach($tags as $element) {
+			if(isset($element->zipcode)) {
+				$item['zipcode'] = $element->zipcode;
+			}
 
-			$element = $element->find('a', 0);
+			if(isset($element->price)) {
 
-			$item = array();
-			$item['uri'] = $element->href;
-			$title = html_entity_decode($element->getAttribute('title'));
-			$content_image = $element->find('div.item_image', 0)->find('.lazyload', 0);
+				$item['price'] = current($element->price);
```

Comment:
You should probably just access the first element in the table.

Reply:
I fix it to do that